plugins/com.google.cloud.tools.eclipse.appengine.ui/src/com/google/cloud/tools/eclipse/appengine/ui/messages.properties

```diff
@@ -1,2 +1 @@
-appengine.runtimes.java7=Java 7, Servlet 2.5 (deprecated)
```

Comment:
How hard would it be to replace the Java 7, Servlet 2.5 runtime with Java 8, Servlet 2.5?

Reply:
I think this requires a bit of restructuring of our code. We use the `AppEngineRuntime` enum constants for this input, and their IDs are used set the `<runtime>` element in `appengine-web.xml`. We could have some ad-hoc conditionals on this, but ideally I think we should change the `AppEngineProjectConfig` interface to have `getServletVersion()`. For this reason and to be on the safe side, I think adding the Servlet 2.5 support should be another PR. Adding a new feature is also independent of deprecating something, and I think there is no reason to combine the two in a single step. However, we can leave the code to generate Servlet 2.5 templates.

Reply:
Make sense. Let's do that in another PR.

Reply:
We can also use the WTP IProjectFacet APIs to query for dynamic web facet support.